File: website/app/src/chardev/backend/entities/SpellEffect.php
<?php

namespace chardev\backend\entities;

// (int)$joinRecord['ProcValue'],
// (int)$joinRecord['Aura'],
// (int)$joinRecord['Effect'],
// (int)$joinRecord['Period'],
// (int)$joinRecord['Value'],
// (int)$joinRecord['Targets'],
// (float)$joinRecord['Coefficient'],
// (int)$joinRecord['Dice'],
// ($joinRecord['Aura'] == 53 ? SpellItemEnchantmentData::getInstance()->fromId($joinRecord['SecondaryEffect']) : (int)$joinRecord['SecondaryEffect']),
// (int)$joinRecord['UsedStat'],
// (int)$joinRecord['ProcChance'],
// (int)$joinRecord['LevelModifier'],
// (int)$joinRecord['ProcSpellID'],
// (int)$joinRecord['ID'],
// (int)$joinRecord['SpellScalingCoefficient']

class SpellEffect {
	
	protected $data = null;
	
	public function __construct( $data ) {
		$this->data = $data;
	}
	
	public function getProcValue() {
		return $this->data[0];
	}
	
	public function getAura() {
		return $this->data[1];
	}
	
	public function getEffect() {
		return $this->data[2];
	}
	
	public function getPeriod() {
		return $this->data[3];
	}
	
	public function getValue() {
		return $this->data[4];
	}
	
	public function getTargets() {
		return $this->data[5];
	}
	
	public function getCoefficient() {
		return $this->data[6];
	}
	
	public function getDice() {
		return $this->data[7];
	}
	
	public function getSecondaryEffect() {
		return $this->data[8];
	}
	
	public function getUsedStat() {
		return $this->data[9];
	}
	
	public function getProcChance() {
		return $this->data[10];
	}
	
	public function getLevelModifier() {
		return $this->data[11];
	}
	
	public function getProcSpellId() {
		return $this->data[12];
	}
	
	public function getId() {
		return $this->data[13];
	}
	
	public function getSpellScalingCoefficient() {
		return $this->data[14];
	}
}

File: tools/cjt/src/org/chardev/cjt/entity/SpellEffect.java
package org.chardev.cjt.entity;

import org.chardev.cjt.entity.factory.LazyField;
import org.chardev.cjt.entity.factory.SpellFactory;

public class SpellEffect {
	protected SpellFactory factory;
	
	protected final int id;
	protected final double procValue;
	protected final int period;
	protected final int value;
	protected final double coefficient;
	protected final double f8;
	protected final int targets;
	protected final int secondaryEffect;
	protected final double procChance;
	protected final int index;
	
	protected LazyField<SpellEffectScaling> scaling;
	
	public SpellEffect( SpellFactory factory, int id, double procValue, int period, int value,
			double coefficient, double f8, int targets, int secondaryEffect,
			double procChance, int index) {
		this.factory = factory;
		
		this.id = id;
		this.procValue = procValue;
		this.period = period;
		this.value = value;
		this.coefficient = coefficient;
		this.f8 = f8;
		this.targets = targets;
		this.secondaryEffect = secondaryEffect;
		this.procChance = procChance;
		this.index = index;
	}

	public int getId() {
		return id;
	}

	public double getProcValue() {
		return procValue;
	}

	public double getPeriod() {
		return period;
	}

	public double getValue() {
		return value;
	}

	public double getCoefficient() {
		return coefficient;
	}

	public double getF8() {
		return f8;
	}

	public int getTargets() {
		return targets;
	}

	public int getSecondaryEffect() {
		return secondaryEffect;
	}

	public double getProcChance() {
		return procChance;
	}

	public int getIndex() {
		return index;
	}
	
	public SpellEffectScaling getScaling() {
		if( null == this.scaling ) {
			this.scaling = new LazyField<SpellEffectScaling>(this.factory.createSpellEffectScaling(this.id));
		}
		
		return this.scaling.value;
	}
}
